# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: docshell/base/WindowContext.h
# Commit: 1cc86398ced0
# Full Hash: 1cc86398ced034fafc66d94799f91221420e1229
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2021-02-18 04:16:06
# Regressor Bug: 1692350
# File Overlap Count: 1
# Description:
#   Bug 1692350 - Implement a telemetry probe to collect page load data for documents that use lazyload r=emilio,chutten
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D104912
# ==============================================================================

diff -r 1ca4c3365fae -r 1cc86398ced0 docshell/base/WindowContext.h
--- a/docshell/base/WindowContext.h	Wed Feb 17 22:02:17 2021 +0000
+++ b/docshell/base/WindowContext.h	Wed Feb 17 22:25:02 2021 +0000
@@ -82,7 +82,10 @@
   FIELD(HasReportedShadowDOMUsage, bool)                               \
   /* Whether the principal of this window is for a local               \
    * IP address */                                                     \
-  FIELD(IsLocalIP, bool)
+  FIELD(IsLocalIP, bool)                                               \
+  /* Whether the corresponding document has `loading='lazy'`           \
+   * images; It won't become false if the image becomes non-lazy */    \
+  FIELD(HadLazyLoadImage, bool)
 
 class WindowContext : public nsISupports, public nsWrapperCache {
   MOZ_DECL_SYNCED_CONTEXT(WindowContext, MOZ_EACH_WC_FIELD)
@@ -175,6 +178,8 @@
 
   bool CanShowPopup();
 
+  bool HadLazyLoadImage() const { return GetHadLazyLoadImage(); }
+
  protected:
   WindowContext(BrowsingContext* aBrowsingContext, uint64_t aInnerWindowId,
                 uint64_t aOuterWindowId, bool aInProcess,
@@ -262,6 +267,9 @@
   bool CanSet(FieldIndex<IDX_IsLocalIP>, const bool& aValue,
               ContentParent* aSource);
 
+  bool CanSet(FieldIndex<IDX_HadLazyLoadImage>, const bool& aValue,
+              ContentParent* aSource);
+
   void DidSet(FieldIndex<IDX_HasReportedShadowDOMUsage>, bool aOldValue);
 
   void DidSet(FieldIndex<IDX_SHEntryHasUserInteraction>, bool aOldValue);